lib/getPostcssResult.js

```diff
@@ -131,17 +162,39 @@ function getCustomSyntax(customSyntax) {
  * @returns {Syntax}
  */
 function cssSyntax(stylelint, filePath) {
-	let fileExtension = '';
-
-	if (filePath) {
-		fileExtension = path
-			.extname(filePath || '')
-			.slice(1)
-			.toLowerCase();
-	}
+	const fileExtension = filePath
+		? path
+				.extname(filePath || '')
+				.slice(1)
+				.toLowerCase()
+		: '';
 
 	const extensions = ['css', 'pcss', 'postcss'];
 
+	switch (fileExtension) {
+		case 'html':
+		case 'markdown':
+		case 'sass':
+			console.warn(`${commonErrorMessages.noConfig(fileExtension)}`);
```

Comment:
Warnings should only be shown, when custom-syntax is not specified.

Reply:
The function containing this `switch` is only reached when the custom syntax is not specified, i.e. the file is going to parsed by either `postcss-safe-parser` or PostCSS built-in CSS parser.

```
const syntax = options.customSyntax
		? getCustomSyntax(options.customSyntax)
		: cssSyntax(stylelint, options.filePath);
```